### PR TITLE
Replace socrata-cloudbees-sbt plugin with new socrata-sbt-plugins

### PR DESCRIPTION
**Pull request by urmilan**
* Fixed all Scalastyle issues
* Removed all redundant plugins
* Fixed geoscript-related issues with publishing

### CODE REVIEW

File: geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    mandatoryInput(name, default, queryParam, _.toString)
+  def mandatoryHeader(name: String, default: Option[String] = None): String =
+    mandatoryInput(name, default, header, _.toString)
+
+  def resourceName: String = mandatoryQueryParam("resourceName")
+  def friendlyName: String = mandatoryQueryParam("friendlyName")
+  def forceLonLat: Boolean = mandatoryInput("forceLonLat", Some(false), queryParam, _.toBoolean)
+  def bypassValidation: Boolean = mandatoryInput("bypassValidation", Some(false), queryParam, _.toBoolean)
+
+  def authToken: String = mandatoryHeader("Authorization")
```
Comment (AEscobarCruz) [inline geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala]: Not sure, but what about making these (resourceName, friendlyName, etc..) lazy vals instead?

Comment (urmilan) [inline geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala]: If we make it a lazy val, it will be evaluated once for the lifetime of the servlet. We want to reevaluate it every time a new request comes in because each request has different parameters and headers.

-----Original Message-----

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/SpatialIndex.scala [hunk 2]
```diff
@@ -113,7 +111,7 @@ object SpatialIndex {
   /**
    * Create a SpatialIndex[String] from a Layer/FeatureSource.  The feature ID will be stored in the index.
    *
-   * @param layer an [[org.geoscript.layer.Layer]] or GeoTools FeatureSource.
+   * @param layer an org.geoscript.layer.Layer or GeoTools FeatureSource.
```
Comment (AEscobarCruz) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/SpatialIndex.scala]: The original reason that I added the function findjar in BuildSettings. When I changed the [[[org.geoscript.layer.layer]] => org.geoscript.layer.layer originally others wanted me to undo and keep the link. So should we revert? I believe we can safely remove the additional (and cumbersome) code from BuildsSettings if we just not link.

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/shapefile/MultiLayerShapefileReader.scala [hunk 3]
```diff
@@ -113,29 +112,28 @@ case class MultiLayerShapefileReader(projectionString: String = ShapeFileConstan
...
+    } yield proj
+
+    contents match {
+      case Success(c) =>
+        Right(c)
+      case Failure(e: Exception) =>
+        logger.warn("\"Reader failed to parse shape layer {}. -> {}", name, e.getMessage)
+        Left(InvalidShapefileSet("Reader failed to parse shape layer '%s'. -> %s".format(name, e.getMessage)))
+      case Failure(e) =>
+        throw e
```
Comment (AEscobarCruz) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/shapefile/MultiLayerShapefileReader.scala]: Very cool!

File: geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala (hunk 1 above)
Comment (AEscobarCruz) [inline geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala]: Gotcha, forgot this isnt rebuilt on new requests.

File: geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/SpatialIndex.scala (hunk 2 above)
Comment (urmilan) [inline geospace-lib/src/main/scala/com/socrata/geospace/lib/regioncache/SpatialIndex.scala]: Oops, I did this when findJars was failing to unblock myself and then forgot to revert. See latest commit :)

**Comment by AEscobarCruz** [comment]
lgtm

File: geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala (hunk 1 above)
Comment (velvia) [inline geospace-http/src/main/scala/com/socrata/geospace/http/GeospaceParams.scala]: I personally wonder if having this style rule makes the code now too obtuse and hard to follow.  Maybe better to just extract out the constants.

**Comment by sokrahta** [comment]
Awesome sauce :+1:
